File: HackerRank/30DaysOfCode/Day20_Sorting.cs
//using System;
//using System.Collections.Generic;
//using System.IO;
//using System.Linq;
//class Solution
//{

//    static void Main(String[] args)
//    {
//        int n = Convert.ToInt32(Console.ReadLine());
//        string[] a_temp = Console.ReadLine().Split(' ');
//        int[] a = Array.ConvertAll(a_temp, Int32.Parse);

//        int arrLen = a.Length - 1;
//        int numSwaps = 0;

//        for (int i = 0; i < n; i++)
//        {

//            for (int j = 0; j < n - 1; j++)
//            {
//                if (a[j] > a[j + 1])
//                {
//                    int tempswap = a[j];
//                    a[j] = a[j + 1];
//                    a[j + 1] = tempswap;
//                    numSwaps++;
//                }
//            }
//            if (numSwaps == 0)
//            {
//                break;
//            }
//        }




//        Console.WriteLine("Array is sorted in {0} swaps.", numSwaps);
//        Console.WriteLine("First Element: {0}", a[0]);
//        Console.WriteLine("Last Element: {0}", a[arrLen]);
//    }

//}

File: java/30DaysOfCoding/Day20_Sorting.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void sort(int[] a, int n){
       int arrLen = a.length - 1;
       int numSwaps = 0;

       for (int i = 0; i < n; i++)
       {

           for (int j = 0; j < n - 1; j++)
           {
               if (a[j] > a[j + 1])
               {
                   int tempswap = a[j];
                   a[j] = a[j + 1];
                   a[j + 1] = tempswap;
                   numSwaps++;
               }
           }
           if (numSwaps == 0)
           {
               break;
           }
       }

       System.out.printf("Array is sorted in %d swaps.\n", numSwaps);
       System.out.printf("First Element: %d\n", a[0]);
       System.out.printf("Last Element: %d\n", a[arrLen]);
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] a = new int[n];
        for(int a_i=0; a_i < n; a_i++){
            a[a_i] = in.nextInt();
        }
        // Write Your Code Here
      sort(a,n);
    }
}
